examples/react-instantsearch/src/Autocomplete.js

```diff
@@ -21,7 +21,7 @@ export function Autocomplete(props) {
     return () => {
       search.destroy();
     };
-  }, []);
+  }, [props]);
```

Comment:
`props` is not a stable value since it's an object. Currently, Autocomplete is destroyed and recreated when you go to a new page for example, maybe we could check only for `props.plugins` (since it's using a `useMemo`), WDYT?

Reply:
That improves the numbers of rerenders indeed. Thanks!

Reply:
While this is a valid concern, relying on `props.plugins` in an example will be misleading to users since it would only re-run the effect when plugins change.

Since this example is designed to be educative, I think we should either stand by this solution not accepting changing props (and we silence the ESLint warning) or we do a deep comparison with `dequal`. What do you think?

Reply:
Good point, for the sake of simplicity, I'll go for the ESLint disable comment for the deps and maybe add a comment explaining why we do this?

Reply:
Having the effect only run once was ok before, but since we now have query suggestions relying on the current category and rendering headers conditionnally, we need autocomplete to rerender accordingly.

I could spread the props in the function signature to single out the plugins so that it would be clearer that these are expected to trigger rerenders, but I don't think removing props from the dependency array altogether is an option.

As for `dequal`, how would it work in this case, would it return early inside the effect?

Reply:
Indeed I didn't take the updated plugins into account. That'll work for now and we can find something else later if it makes sense.